### PR TITLE
Bug 1854354 - Bookmarks TestRail matching

### PR DESCRIPTION
**Pull request by AndiAJ**
These changes are part of our data alignment between testRail and our automated UI tests.

For more information please see this [document](https://docs.google.com/document/d/1lRN75vjmqFHqT4aG21JPSriQbUSahCxJplWSi2ec-lw/edit?usp=sharing)

Deleted the following UI tests that don't have any coresponding manual test in TestRail:
- `verifyBookmarkButtonTest` 
- `addBookmarkTest`
- `multipleBookmarkDeletionsTest`
- `verifyCloseMenuTest` 

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1854354
https://bugzilla.mozilla.org/show_bug.cgi?id=1854566

### CODE REVIEW

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/BookmarksTest.kt [hunk 1]
```diff
@@ -116,54 +104,7 @@ class BookmarksTest {
...
-        }.enterURLAndEnterToBrowser(defaultWebPage.url) {
-        }.openThreeDotMenu {
-        }.bookmarkPage {
-        }.openThreeDotMenu {
-            verifyEditBookmarkButton()
-        }
-    }
-
-    @Test
-    fun addBookmarkTest() {
```
Comment (sv-ohorvath) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/ui/BookmarksTest.kt:133]: verifyBookmarkButtonTest and addBookmarkTest could be merged into one and add a TR test for them.

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/BookmarksTest.kt [hunk 2]
```diff
@@ -513,9 +449,10 @@ class BookmarksTest {
         }
     }
 
+    // TestRail link: https://testrail.stage.mozaws.net/index.php?/cases/view/325644
     @SmokeTest
     @Test
-    fun deleteMultipleSelectionTest() {
+    fun deleteMultipleSelectedBookmarksTest() {
```
Comment (sv-ohorvath) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/ui/BookmarksTest.kt:455]: Let's keep just this and openMultipleSelectedBookmarksInANewTabTest as smoke tests, you can remove the tag from openMultipleSelectedBookmarksInPrivateTabTest and the other multiple selection tests.

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/BookmarksTest.kt [hunk 3]
```diff
@@ -598,43 +522,10 @@ class BookmarksTest {
...
-        }.openBookmarks {
-            verifyFolderTitle("3")
-        }
-    }
-
+    // TestRail link: https://testrail.stage.mozaws.net/index.php?/cases/view/325639
     @SmokeTest
     @Test
-    fun changeBookmarkParentFolderTest() {
+    fun createBookmarkFolderTest() {
```
Comment (sv-ohorvath) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/ui/BookmarksTest.kt:528]: let's remove the Smoke tag from this one as well

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/BookmarksTest.kt [hunk 4]
```diff
@@ -431,9 +361,13 @@ class BookmarksTest {
...
+        }.openThreeDotMenu(defaultWebPage.title) {
+        }.clickDelete {
+            verifyBookmarkIsDeleted(defaultWebPage.title)
         }
     }
 
+    // TestRail link: https://testrail.stage.mozaws.net/index.php?/cases/view/2300275
     @SmokeTest
     @Test
     fun bookmarksMultiSelectionToolbarItemsTest() {
```
Comment (sv-ohorvath) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/ui/BookmarksTest.kt:373]: Remote the Smoke tag from this one too.

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/BookmarksTest.kt [hunk 5]
```diff
@@ -177,9 +118,10 @@ class BookmarksTest {
         }
     }
 
+    // TestRail link: https://testrail.stage.mozaws.net/index.php?/cases/view/2299619
     @SmokeTest
     @Test
-    fun cancelEditBookmarkTest() {
+    fun cancelingChangesInEditModeAreNotSavedTest() {
```
Comment (sv-ohorvath) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/ui/BookmarksTest.kt:124]: No need to keep the Smoke tag here, this is a low-priority test.

Comment (AndiAJ) [reply fenix/app/src/androidTest/java/org/mozilla/fenix/ui/BookmarksTest.kt:147]: Done !

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/BookmarksTest.kt (hunk 4 above)
Comment (AndiAJ) [reply fenix/app/src/androidTest/java/org/mozilla/fenix/ui/BookmarksTest.kt:395]: Done !

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/BookmarksTest.kt (hunk 2 above)
Comment (AndiAJ) [reply fenix/app/src/androidTest/java/org/mozilla/fenix/ui/BookmarksTest.kt:476]: Done !

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/BookmarksTest.kt (hunk 3 above)
Comment (AndiAJ) [reply fenix/app/src/androidTest/java/org/mozilla/fenix/ui/BookmarksTest.kt:548]: Done !

**Review by isabelrios — APPROVED** [review]
LGTM! Great work with all these changes, thanks!

**Review by sv-ohorvath — APPROVED** [review]

**Pull request merged**